### PR TITLE
ContextManagersTest: Merge the test from MakeTempTest

### PR DESCRIPTION
**Pull request by souravsingh**
This patch merges the test case written in MakeTempTest.py into the script.

Fixes https://github.com/coala-analyzer/coala/issues/1416

### CODE REVIEW

**Comment by souravsingh** [comment]
@Adrianzatreanu This is the PR written. Waiting for all tests to pass

**Comment by Makman2** [comment]
Here too, please add `Fixes ...` to commit body^^

**Comment by Makman2** [comment]
@souravsingh ping

**Comment by souravsingh** [comment]
pong

**Comment by Makman2** [comment]
@souravsingh you need help? this PR shouldn't be that complex, and then we can merge it :)

**Comment by souravsingh** [comment]
Is it good for merge?

File: tests/misc/ContextManagersTest.py [hunk 1]
```diff
@@ -108,6 +108,15 @@ def test_make_temp(self):
...
             self.assertTrue(os.path.basename(f_b).startswith("pre"))
 
+    def test_temp_file_existence(self):
+        """
+        Test that the temporary file created exists only within the with
+        statement context and not outside it
+        """
+        with make_temp() as temporary:
+            self.assertTrue(os.path.isfile(temporary))
+        self.assertFalse(os.path.isfile(temporary))
```
Comment (Makman2) [inline tests/misc/ContextManagersTest.py]: You should move `self.assertFalse(...)` to `test_make_temp` above, below the first `with`-statement, then you can delete the rest :)

Comment (Makman2) [inline tests/misc/ContextManagersTest.py]: And please remove `MakeTempTest.py` then^^

**Comment by Adrianzatreanu** [comment]
it is passing like no tests, and it is outdated. so no, it's not good for merge yet :/ but you're getting there!

File: tests/misc/ContextManagersTest.py [hunk 2]
```diff
@@ -103,20 +103,12 @@ def test_make_temp(self):
         with make_temp() as f_a:
             self.assertTrue(os.path.isfile(f_a))
             self.assertTrue(os.path.basename(f_a).startswith("tmp"))
+            self.assertFalse(os.path.isfile(temporary))
```
Comment (Makman2) [inline tests/misc/ContextManagersTest.py]: that belongs outside the `with` stmt^^

**Comment by Adrianzatreanu** [comment]
@souravsingh are you still working on this? feel free to message us on gitter :)

**Comment by Makman2** [comment]
Taking this over now.

**Comment by Makman2** [comment]
https://github.com/coala-analyzer/coala/pull/2120 takes over.

**Pull request closed without merging**